handlers/messangi/messangi.go

```diff
@@ -0,0 +1,132 @@
+package messangi
+
+import (
+	"context"
+	"crypto/hmac"
+	"crypto/sha256"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/nyaruka/courier"
+	"github.com/nyaruka/courier/handlers"
+	"encoding/base64"
+	"encoding/hex"
+	"github.com/nyaruka/courier/utils"
+	"encoding/xml"
+)
+
+const (
+	configPublicKey  = "messangi_public_key"
+	configPrivateKey = "messangi_private_key"
+	configInstanceId = "messangi_instance_id"
+	configCarrierId  = "messangi_carrier_id"
```

Comment:
no need to have these constants start with `messangi`, a channel can only be of a single type at a time.